src/group_theory/order_of_element.lean

```diff
@@ -684,3 +688,28 @@ subgroup_of_idempotent (S ^ (fintype.card G)) ⟨1, one_mem⟩ begin
 end
 
 end pow_is_subgroup
+
+section linear_ordered_ring
+
+variable [linear_ordered_ring G]
+
+lemma order_of_abs_ne_one (h : |x| ≠ 1) : order_of x = 0 :=
+begin
+  rw order_of_eq_zero_iff',
+  intros n hn hx,
+  replace hx : |x| ^ n = 1 := by simpa only [abs_one, abs_pow] using congr_arg abs hx,
+  cases h.lt_or_lt with h h,
+  { exact ((pow_lt_one (abs_nonneg x) h hn.ne').ne hx).elim },
+  { exact ((one_lt_pow h hn.ne').ne' hx).elim }
+end
+
+lemma linear_ordered_ring.order_of_le_two : order_of x ≤ 2 :=
+begin
+  cases ne_or_eq (|x|) 1 with h h,
```

Comment:
```suggestion
  by_cases h : |x| = 1,
```
optional, but `ne_or_eq` isn't doing any terribly advanced case analysis here

Reply:
this is just for the order; if I did `by_cases` there's no way to have the "boring" part first